src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -507,6 +507,10 @@ void PolicyHandler::OnDeviceConsentChanged(const std::string& device_id,
       policy_manager_->ReactOnUserDevConsentForApp(policy_app_id, is_allowed);
 
       policy_manager_->SendNotificationOnPermissionsUpdated(policy_app_id);
+
+      if (policy_manager_->IsPredataPolicy(policy_app_id) && !is_allowed) {
```

Comment:
when will this condition eval to true? a few lines above the same condition will lead to a continue statement

do we need to update the heartbeat timeout again when an app leaves predata consent and is allowed?